tough/src/lib.rs

```diff
@@ -215,15 +216,15 @@ impl<'a, T: Transport> Repository<'a, T> {
     ///
     /// `metadata_base_url` and `targets_base_url` are the HTTP(S) base URLs for where the client
     /// can find metadata (such as root.json) and targets (as listed in targets.json).
-    pub fn load<R: Read>(transport: &'a T, settings: Settings<R>) -> Result<Self> {
+    pub fn load<R: Read>(transport: Box<dyn Transport>, settings: Settings<R>) -> Result<Self> {
```

Comment:
I think you may be able to simplify the interface by taking `Into<Box<dyn Transport>>` because Box has a blanket impl for any T (that just boxes it) so you can cal `.into()` inside this function to box the input if it's not already.  Then this would accept `Transport` or `Box<Transport>`.

(I'm thinking this would cut out half of the changed files from this commit, which are just adding `Box::new` to load calls...)

Reply:
I'm still deciding whether it's worth it to make this change to the early commit named `remove type parameter for transport` since that is a transient commit anyway.

But where I should, and definitely will, implement this change is in the commit named `simplify repository load` where the `Box<dyn Transport>` is a member of `Options`.